src/main/java/org/gridsuite/actions/server/ContingencyListService.java

```diff
@@ -346,6 +350,7 @@ public ScriptContingencyList replaceFilterContingencyListWithScript(UUID id) {
         return filter.map(entity -> {
             String script = generateGroovyScriptFromFilters(fromFilterContingencyListEntityAttributes(entity));
             var scriptContingencyListEntity = new ScriptContingencyListEntity(new ScriptContingencyList(id, entity.getName(), script, entity.getDescription()));
+            scriptContingencyListEntity.setId(id);
```

Comment:
Do they absolutly need to have the same ID ? I mean it could create two elements with the same UUID if the next save or delete fail... If you replace Filter with script contingency type, maybe it should be another generated UUID

Reply:
Actually I think the `setID` accessor should be forbidden, and the `UUID.randomUUID()` should be called in the constructor of each entity